packages/contracts-bedrock/scripts/Deployer.sol

```diff
@@ -561,15 +563,44 @@ abstract contract Deployer is Script {
     }
 
     /// @dev Returns the value of the internal `_initialized` storage slot for a given contract.
-    function loadInitializedSlot(string memory _contractName, bool _isProxy) public returns (uint8 initialized_) {
-        StorageSlot memory slot = getInitializedSlot(_contractName);
-        if (_isProxy) {
-            _contractName = string.concat(_contractName, "Proxy");
+    function loadInitializedSlot(string memory _contractName) public returns (uint8 initialized_) {
+        uint256 contractNameLength = bytes(_contractName).length;
+        address addr;
+        // Check if the contract name ends with `Proxy` and if so, get the implementation address
+        string memory suffix = _substring(_contractName, contractNameLength - 5, contractNameLength);
+        if (bytes(suffix).length == bytes("Proxy").length && keccak256(bytes(suffix)) == keccak256(bytes("Proxy"))) {
```

Comment:
We now have an implicit requirement that proxy contracts must be named `*Proxy`. We should make this requirement explicit by documenting it in the style guide, and ideally trying to automate this check (e.g. semgrep rule that if `Proxy` is found in a contract name it must be the suffix) but that is tricky to do generically

Reply:
Ah thanks, I didn't know that. Still agree with my recommendations, but happy to have them done in a separate PR

Reply:
I agree with your recommendation as well, making it explicit is better